weathervane.pl

```diff
@@ -654,6 +680,34 @@ sub getComputeResourceForInstance {
 my $workloadNum = 1;
 my @workloads;
 foreach my $workloadParamHashRef (@$workloadsParamHashRefs) {
+	# Overwrite the workload's parameters with those specified by the configuration size.
+	my $configSize = $workloadParamHashRef->{'configurationSize'};
+	if ($configSize ne "custom") {
+		my @configKeys = keys $fixedConfigs;
+		if (!($configSize ~~ @configKeys)) {
+			$console_logger->error("Error: For workload " . $workloadNum 
+				.  " configurationSize " . $configSize . " does not exist.");
+		}
+		my $config = $fixedConfigs->{$configSize};
+		foreach my $key (keys %$config) {
+			if ($key eq "numDrivers") {
+				# If the user set the number of driver, then use that value.
+				# Otherwise the number of drivers is the number in the
+				# fixed config times the number of appInstances
+				if (!$workloadParamHashRef->{"numDrivers"} && ($#{$workloadParamHashRef->{"drivers"}} < 0)) {
+					my $numAppInstances = $workloadParamHashRef->{"numAppInstances"};
+					if (!$numAppInstances) {
+						my $instancesListRef = $workloadParamHashRef->{"appInstances"};
+						$numAppInstances = $#{$instancesListRef} + 1;					
+					}
+					$workloadParamHashRef->{$key} = $config->{$key} * $numAppInstances;
+				}
+			} else {
+				$workloadParamHashRef->{$key} = $config->{$key};
```

Comment:
I have created a Jira issue for this.  It isn't a straightforward change because we need to determine whether the value being overwritten is something the user set in the config file (or command-line) or is just the default.  We don't want to print a message when we override the default.